core/services/src/main/java/org/exoplatform/services/cms/link/impl/LinkManagerImpl.java

```diff
@@ -444,4 +444,30 @@ public void updateSymlink(Node node) throws RepositoryException {
     }
   }
 
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public Node getNodeSymlinksUnderFolder(String contentUUID, String folderPath, String workspace) {
+    ManageableRepository repository = WCMCoreUtils.getRepository();
+    SessionProviderService sessionProviderService = CommonsUtils.getService(SessionProviderService.class);
+    SessionProvider sessionProvider = sessionProviderService.getSystemSessionProvider(null);
+    try {
+      Session session = sessionProvider.getSession(workspace, repository);
+      QueryManager queryManager = session.getWorkspace().getQueryManager();
+      String queryString = new StringBuilder("select * from exo:symlink ").append(" where exo:uuid = '").
+              append(contentUUID).append("' AND jcr:path like '%").
+              append(folderPath).append("/%' order by exo:dateCreated DESC").toString();
+      Query query = queryManager.createQuery(queryString, Query.SQL);
+      QueryResult queryResult = query.execute();
+      NodeIterator iterator = queryResult.getNodes();
+      while (iterator.hasNext()) {
+        return iterator.nextNode();

```

Comment:
We may have multiple symlinks of the same content, let's return a list of nodes

Reply:
Is it possible to have multiple symlinks of the same content in the same path? why we don't just add LIMIT 1 to the sql query if it's possible ? wdyt?

Reply:
Yes we may have multiple symlinks inside a folder that targets one document, and of course we can have other symlinks inside child folder of that folder

Reply:
done